src/components/map/Map.tsx

```diff
@@ -157,138 +107,219 @@ export const Map = ({
+    });
+
+    const points = locations.map((location, index) => ({
+      type: 'Feature',
+      properties: { cluster: false, id: location.id, index },
+      geometry: {
+        type: 'Point',
+        coordinates: [location.position.longitude, location.position.latitude],
+      },
+    }));
+
+    index.load(points);
+    const bounds = [-180, -85, 180, 85];
+    setClusters(index.getClusters(bounds, zoomLevel));
+
+    // Berechne die Begrenzungskoordinaten für alle Marker
+    if (locations.length > 0 && isInitialFit) {
+      const coordinates = locations.map(loc => [loc.position.longitude, loc.position.latitude]);
+      if (locations.length === 1) {
+        if (cameraRef.current) {
+          cameraRef.current.setCamera({
+            centerCoordinate: coordinates[0],
+            zoomLevel: 15,
+            animationDuration: 0,
+          });
+          setTimeout(() => {
+            setIsLoading(false);
+          }, 4000); // Timeout in Millisekunden anpassen je nach Bedarf
+          setIsInitialFit(false); // Verhindert zukünftige automatische Anpassungen
+        }
+      } else {
+        const [minLng, minLat] = coordinates.reduce((prev, curr) => [Math.min(prev[0], curr[0]), Math.min(prev[1], curr[1])], [Infinity, Infinity]);
+        const [maxLng, maxLat] = coordinates.reduce((prev, curr) => [Math.max(prev[0], curr[0]), Math.max(prev[1], curr[1])], [-Infinity, -Infinity]);
+        const deltaLng = (maxLng - minLng) * 0.1; // Add 10% padding to the bounds
+        const deltaLat = (maxLat - minLat) * 0.1; // Add 10% padding to the bounds
+        if (cameraRef.current) {
+          cameraRef.current.fitBounds(
+            [minLng - deltaLng, minLat - deltaLat],
+            [maxLng + deltaLng, maxLat + deltaLat],
+            0 // animationDuration in ms
+          );
+          setTimeout(() => {
+            setIsLoading(false);
+          }, 1000); // Timeout in Millisekunden anpassen je nach Bedarf
+          setIsInitialFit(false); // Verhindert zukünftige automatische Anpassungen
+        }
+      }
+    }
+  }, [locations, clusterDistance, zoomLevel, isInitialFit]);
+
+  const handleMyLocationButtonPress = () => {
+    if (refForMapView.current && locationSettings?.currentLocation) {
+      cameraRef.current.setCamera({
+        centerCoordinate: [
+          locationSettings.currentLocation.latitude,
+          locationSettings.currentLocation.longitude
+        ],
+        zoomLevel: 15,
+        animationDuration: 1000
+      });
+    }
+  };
+
   return (
     <View style={[styles.container, style]}>
-      <MapView
-        clusterColor={colors.primary}
-        clusterFontFamily="regular"
-        clusteringEnabled={clusteringEnabled}
-        radius={clusterDistance}
-        renderCluster={renderCluster}
-        initialRegion={initialRegion}
-        region={updatedRegion}
-        mapType={device.platform === 'android' ? MAP_TYPES.NONE : MAP_TYPES.STANDARD}
+      <LoadingSpinner loading={isLoading} />
```

Comment:
How about using it the same way we use it in other parts of the code to preserve the structure in the code?

With an if check before return. For example:

```
  if (isLoading) {
    return <LoadingSpinner loading/>;
  }
```

Reply:
imo its more efficient to use it this way since it requires less code and the best code is no code therefore I'd leave it as it is
https://github.com/smart-village-solutions/smart-village-app-app/pull/1154/commits/7d15a3d5f2af482f630518fdc922da1732420e58